Examples/Pass/ticketLockNonAtomicRelease2.cvf

```diff
@@ -0,0 +1,55 @@
+/*
+ * Linux-style ticketed lock.
+ */
+
+global int ticket;  // The next ticket to hand out.
+global int serving; // The current ticket holding the lock.
+local int t;  // The thread's current ticket.
+local int s;  // The thread's current view of serving.
+
+/*
+ * Locks the ticket lock.
+ */
+method lock() {
+  {| emp |}
+    <t = ticket++>;
+  {| holdTick(t) |}
+    do {
+      {| holdTick(t) |}
+        <s = serving>;
+      {| if s == t then holdLock() else holdTick(t) |}
+    } while (s != t);
+  {| holdLock() |}
+}
+
+/*
+ * Unlocks the ticket lock.
+ */
+method unlock() {
+  {| holdLock() |}
+    <s = serving>;
+  {| /*relLock(s)*/ holdLock() |}
```

Comment:
This looks sound to me, but for unexpected reasons.

`{| holdLock() |} <serving = (s + 1)> {| emp |}` looks ok, because `emp` is constrained to `true` in this version.  This is an odd choice for `emp`, but came from HSF.

Since we're not constraining the value of `serving` afterwards, we can get to various different situations, all of which seem sound:
* `ticket >= (s + 1) > serving`, which is expected behaviour and what we want it to be, morally.
* `(s + 1) == serving`, ie don't actually give up the lock at all.  Sound because we never actually state that we got rid of the lock, just that we forgot we had one.  This will deadlock (nobody who has a ticket can ever get the lock), but we're only showing partial correctness, so this goes through.
* `serving > ticket`, which is a surprising turn of affairs, but is also valid: nobody who has a ticket will ever be served, and locking will deadlock (the do-while loop will spin forever), but that's ok because we never explicitly rely on this.

The bottom line is that `serving <= ticket` in this case is actually a problem/liveness constraint, not a proof one, and without it HSF merrily proves programs that will deadlock but satisfy their specifications.

The more interesting case is if we actually move `serving` such that `serving > ticket`.  This is obviously bad behaviour, but HSF here inferred that in terms of partial correctness this doesn't really matter: we can